importer/Uploader.py

```diff
@@ -159,6 +159,10 @@ def add_claims(self, wd_item, claims, log):
             for claim in claims:
                 prop = claim
                 for x in claims[claim]:
+                    if (x.get("if_empty") and
+                            self.item_has_prop(x["prop_name"], wd_item)):
```

Comment:
You need to pass the name of the property here, so this won't work (because `x["prop_name"]` gives a KeyError).

A way to fix this might be modifying `item_has_prop()` so that it can take either a P-id or a property name, and then you could use `self.item_has_prop(prop, wd_item)`.

Reply:
That is, if I understand correctly what you expected `x["prop_name"]` to return.

Reply:
So does `prop` and `claims[prop]["prop_name"]` both contain Pid?

Reply:
`prop` is a Pid.

`claims[prop]` looks like this:

`[{'quals': {}, 'refs': [{'source': {'value': 'Q28563569', 'prop': 'P248'}, 'reference_url': {'value': 'https://tools.wmflabs.org/heritage/api/api.php?action=search&format=json&srcountry=mt&srlanguage=de&srid=146', 'prop': 'P854'}, 'published': {'value': {'month': 9, 'day': 14, 'year': 2017}, 'prop': 'P577'}}], 'value': 'Q19685249', 'if_empty': False}]`

Reply:
Thanks. new patch up.

And will rebase